code/modules/reagents/reagents.dm

```diff
@@ -215,24 +215,28 @@
 	return null
 
 // Addiction
-/datum/reagent/proc/addiction_act_stage1(mob/living/carbon/M)
-	if(prob(30))
+// Addiction
+/datum/reagent/proc/addiction_act_stage1(mob/living/carbon/human/M)
+	if(prob(30) && ishuman(M))
 		to_chat(M, SPAN_NOTICE("You feel like having some [name] right about now."))
 
-/datum/reagent/proc/addiction_act_stage2(mob/living/carbon/M)
+/datum/reagent/proc/addiction_act_stage2(mob/living/carbon/human/M)
 	if(prob(30))
 		to_chat(M, SPAN_NOTICE("You feel like you need [name]. You just can't get enough."))
 
-/datum/reagent/proc/addiction_act_stage3(mob/living/carbon/M)
+/datum/reagent/proc/addiction_act_stage3(mob/living/carbon/human/M)
 	if(prob(30))
 		to_chat(M, SPAN_DANGER("You have an intense craving for [name]."))
+		M.sanity.changeLevel(-5)
 
-/datum/reagent/proc/addiction_act_stage4(mob/living/carbon/M)
+/datum/reagent/proc/addiction_act_stage4(mob/living/carbon/human/M)
 	if(prob(30))
 		to_chat(M, SPAN_DANGER("You're not feeling good at all! You really need some [name]."))
+		M.sanity.changeLevel(-10)
```

Comment:
That may (and very likely will) cause people to go insane if they forgot to put another cig in their mouth after old one is gone. Only annoyance.

Reply:
maybe make nicotime unaddictable?

Reply:
![image](https://user-images.githubusercontent.com/49622619/152575293-bc01fa90-cc08-4198-93cd-0d5cce8f3713.png)
no we cant do it now

Reply:
> > maybe make nicotime unaddictable?
> 
> Nah, nicotine is very addictive.

then its intended; yet another smoking nerf!